js/private/js_library.bzl

```diff
@@ -163,25 +172,35 @@ target in {file_basename}'s package and add that target to the deps of {this_tar
 def _js_library_impl(ctx):
     is_windows = ctx.target_platform_has_constraint(ctx.attr._windows_constraint[platform_common.ConstraintValueInfo])
 
-    sources, declarations = _gather_sources_and_declarations(
+    sources, declarations_from_sources = _gather_sources_and_declarations(
         ctx = ctx,
         targets = ctx.attr.srcs,
         files = ctx.files.srcs,
         is_windows = is_windows,
     )
 
+    sources_from_declarations, declarations = _gather_sources_and_declarations(
+        ctx = ctx,
+        targets = ctx.attr.declarations,
+        files = ctx.files.declarations,
+        is_windows = is_windows,
+    )
+
+    sources = depset(transitive = [sources, sources_from_declarations])
+    declarations = depset(transitive = [declarations, declarations_from_sources, sources_from_declarations])
```

Comment:
`declarations_from_sources` alongside `sources_from_declarations` is hurting by brain (today at least) - can you bikeshed on them a bit more?

Reply:
Hard to find namings that read easy here, changed to:
```
    sources, declarations = _gather_sources_and_declarations(
        ctx = ctx,
        targets = ctx.attr.srcs,
        files = ctx.files.srcs,
        is_windows = is_windows,
    )

    additional_sources, additional_declarations = _gather_sources_and_declarations(
        ctx = ctx,
        targets = ctx.attr.declarations,
        files = ctx.files.declarations,
        is_windows = is_windows,
    )

    sources = depset(transitive = [sources, additional_sources])
    declarations = depset(transitive = [declarations, additional_sources, additional_declarations])
```